postgres/pgdump.go

```diff
@@ -185,13 +185,48 @@ func processStatements(conv *internal.Conv, statements []nodes.Node) *copyOrInse
 			if conv.SchemaMode() {
 				processVariableSetStmt(conv, n)
 			}
+		case nodes.IndexStmt:
+			if conv.SchemaMode() {
+				processIndexStmt(conv, n)
+			}
 		default:
 			conv.SkipStatement(prNodes([]nodes.Node{node}))
 		}
 	}
 	return nil
 }
 
+func toSchemaIndexKeys(indexParams nodes.List) (keys []schema.Key) {
+	var schemaKeys []schema.Key
+	for _, ip := range indexParams.Items {
+		schemaKeys = append(schemaKeys, schema.Key{Column: *ip.(nodes.IndexElem).Name})
+	}
+	return schemaKeys
+}
+
+func processIndexStmt(conv *internal.Conv, n nodes.IndexStmt) {
+	if n.Relation == nil {
+		logStmtError(conv, n, fmt.Errorf("relation is nil"))
+		return
+	}
+	tableName, err := getTableName(conv, *n.Relation)
+	if err != nil {
+		logStmtError(conv, n, fmt.Errorf("can't get table name: %w", err))
+		return
+	}
+	if _, ok := conv.SrcSchema[tableName]; ok {
+		ctable := conv.SrcSchema[tableName]
+		var cols []string
+		for _, ip := range n.IndexParams.Items {
+			cols = append(cols, *ip.(nodes.IndexElem).Name)
+		}
+		ctable.Indexes = append(ctable.Indexes, schema.Index{Name: *n.Idxname, Unique: *&n.Unique, Keys: toIndexKeys(conv, tableName, n.IndexParams.Items)})
+		conv.SrcSchema[tableName] = ctable
+	} else {
+		conv.SkipStatement(prNodes([]nodes.Node{n}))
+	}
```

Comment:
I wonder if this is an expected error scenario? If not, should be use `conv.Unexpected` instead with appropriate message?

Reply:
whenever there is a situation when we can not read a statement we report skipStatement for that. but we also should report unexpected for that, i'll add it.

Reply:
unexpected statement also added